pkg/packetfilter/nftables/nftables.go

```
@@ -0,0 +1,535 @@
+		return errors.Wrapf(err, "failed to check if %q chain exists", chain.Name)
+	}
+
+	if exists {
+		return nil
+	}
+
+	tx := p.nftables.NewTransaction()
+	tx.Add(&knftables.Table{
+		Comment: ptr.To("rules for submariner"),
+	})
+
+	tx.Add(&knftables.Chain{
+		Name: chain.Name,
+	})
+
+	err = p.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrapf(err, "error creating chain %q", chain.Name)
+}
+
+func (p *packetFilter) DeleteIPHookChain(chain *packetfilter.ChainIPHook) error {
+	return p.deleteChain(chain.Name)
+}
+
+func (p *packetFilter) DeleteChain(_ packetfilter.TableType, chain string) error {
+	return p.deleteChain(chain)
+}
+
+func (p *packetFilter) deleteChain(chain string) error {
+	tx := p.nftables.NewTransaction()
+	tx.Delete(&knftables.Chain{
+		Name: chain,
+	})
+
+	err := p.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error deleting chain %q", chain)
+}
+
+func (p *packetFilter) ClearChain(_ packetfilter.TableType, chain string) error {
+	tx := p.nftables.NewTransaction()
+	tx.Flush(&knftables.Chain{
+		Name: chain,
+	})
+
+	err := p.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error clearing chain %q", chain)
+}
+
+func (p *packetFilter) Delete(_ packetfilter.TableType, chain string, rule *packetfilter.Rule) error {
+	kRule := p.getNftablesRuleFromList(chain, rule)
+	if kRule == nil {
+		return nil
+	}
+
+	tx := p.nftables.NewTransaction()
+	tx.Delete(&knftables.Rule{
+		Chain:  chain,
+		Handle: kRule.Handle,
+	})
+
+	err := p.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error deleting rule %q from chain %q", *kRule.Comment, chain)
+}
+
+func (p *packetFilter) getNftablesRuleFromList(chain string, rule *packetfilter.Rule) *knftables.Rule {
+	existingRules, err := p.nftables.ListRules(context.TODO(), chain)
+	if err != nil {
+		return nil
```

This should return the error instead of silencing it.